Java-Control-Statements.md

```diff
@@ -0,0 +1,216 @@
+the value of an expression. It is actually  a better alternative than using large `if-else statements`.
+
+**Example :**
+
+```
+public class Test {
+
+   public static void main(String args[])
+      {
+    char grade = args[0].charAt(0);
+    char grade = 'C';
+
+      switch(grade)
+      {
+         case 'A' :
+            System.out.println("Excellent!"); 
+            break;
+         case 'B' :
+            System.out.println("Good");
+            break;
+         case 'D' :
+            System.out.println("Passed");
+         case 'F' :
+            System.out.println("Failed");
+            break;
+         default :
+            System.out.println("Invalid grade");
+      }
+      System.out.println("Your grade is " + grade);
+   }
+}
+```
+
+**Output :**
+
+```
+$ java Test 
+Well done 
+Your grade Good
+```
+
+Now Compile and run above program using various [command line arguments](https://docs.oracle.com/javase/tutorial/essential/environment/cmdLineArgs.html). Click on the links for using `command line arguments` in [Eclipse](http://www.cs.colostate.edu/helpdocs/eclipseCommLineArgs.html) and [Netbeans](http://netbeanside61.blogspot.in/2009/02/using-command-line-arguments-in.html).
+
+```
+$ java Test "A"
+Well done
+Your grade Excellent!
+```
+
+In `switch statement` the value of the expression is compared with each of the cases. If a match is found , the code sequence
+following the case is executed. If none of the expressions match then the `default statement` is executed. The `default statement` is
+optional and if no default is provided, then no further action is taken. The `break statement` is used to terminate a statement
+sequenc. The `break statement` is optional. However, if you omit the `break` execution will continue to the next statement.
+
+**Example :**
+
+```
+public class Test {
+   public static void main(String args[])
+       {
+    char grade = args[0].charAt(0);
+    char grade = 'C';
+
+      switch(grade)
+      {
+         case 'A' :
+            System.out.println("Excellent!"); 
+         case 'B' :
+            System.out.println("Good");
+         case 'D' :
+            System.out.println("Passed");
+         case 'F' :
+            System.out.println("Failed");
+            break;
+         default :
+            System.out.println("Invalid grade");
+      }
+      System.out.println("Your grade is " + grade);
+   }
+}
+```
+
+**Output :**
+
+```
+$ java Test "A"
+Well done
+Your grade Excellent!
+Your grade Good
+Your grade Passed
+Your grade Failed
+```
+
+
+###Summary:
+
+The `switch` differs from the `if` in that `switch` can only test for equality, whereas if can evaluate any type of Boolean expressio
+n. Also, note that two case constants can't have identical values in same `switch`.
+A `switch statement` is usually more efficient than a set of `nested ifs`. In case of `if-else` the Java-compiler would have a check,
```

Comment:
This is debatable. So present at least two sides of the argument. What do you mean by **efficient**? Runtime performance in terms of speed or memory consumed?